### ISSUE TITLE
The image fails to build

### ISSUE THREAD

**Issue by alejandro-perez** [iay/md-validator#7]
When I try to build the docker image, I'm getting the following error:
```
tep 7/17 : RUN mvn --batch-mode     -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn     package
 ---> Running in 345fe109d37c
[INFO] Scanning for projects...
[INFO] 
[INFO] ----------------< uk.org.iay.md.validator:md-validator >----------------
[INFO] Building Metadata Validation Micro-Service 0.1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (maven-version) @ md-validator ---
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:copy-resources (copy-resources) @ md-validator ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- swagger-codegen-maven-plugin:2.3.1:generate (default) @ md-validator ---
[INFO] reading from swagger/swagger.yaml
[INFO] Set base package to invoker package (uk.org.iay.md.validator)
[INFO] writing file /application/target/generated-sources/swagger/src/main/java/uk/org/iay/md/validator/models/InlineResponse404.java
[INFO] writing file /application/target/generated-sources/swagger/src/main/java/uk/org/iay/md/validator/models/Status.java
[INFO] writing file /application/target/generated-sources/swagger/src/main/java/uk/org/iay/md/validator/models/Validator.java
[INFO] Skipped generation of /application/target/generated-sources/swagger/src/main/java/uk/org/iay/md/validator/api/ValidatorsApiController.java due to rule in .swagger-codegen-ignore
[INFO] Skipped generation of /application/target/generated-sources/swagger/src/main/java/uk/org/iay/md/validator/api/ValidatorsApi.java due to rule in .swagger-codegen-ignore
[INFO] Skipped generation of /application/target/generated-sources/swagger/pom.xml due to rule in .swagger-codegen-ignore
[INFO] Skipped generation of /application/target/generated-sources/swagger/README.md due to rule in .swagger-codegen-ignore
[INFO] writing file /application/target/generated-sources/swagger/src/main/java/io/swagger/configuration/HomeController.java
[INFO] writing file /application/target/generated-sources/swagger/src/main/java/uk/org/iay/md/validator/Swagger2SpringBoot.java
[INFO] writing file /application/target/generated-sources/swagger/src/main/java/uk/org/iay/md/validator/RFC3339DateFormat.java
[INFO] Skipped generation of /application/target/generated-sources/swagger/src/main/resources/application.properties due to rule in .swagger-codegen-ignore
[INFO] Skipped generation of /application/target/generated-sources/swagger/src/main/java/uk/org/iay/md/validator/api/ApiException.java due to rule in .swagger-codegen-ignore
[INFO] writing file /application/target/generated-sources/swagger/src/main/java/uk/org/iay/md/validator/api/ApiResponseMessage.java
[INFO] Skipped generation of /application/target/generated-sources/swagger/src/main/java/uk/org/iay/md/validator/api/NotFoundException.java due to rule in .swagger-codegen-ignore
[INFO] writing file /application/target/generated-sources/swagger/src/main/java/uk/org/iay/md/validator/api/ApiOriginFilter.java
[INFO] writing file /application/target/generated-sources/swagger/src/main/java/io/swagger/configuration/SwaggerDocumentationConfig.java
[INFO] writing file /application/target/generated-sources/swagger/.swagger-codegen/VERSION
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ md-validator ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 35 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ md-validator ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 18 source files to /application/target/classes
[ERROR] error reading /root/.m2/repository/org/bouncycastle/bcprov-jdk15on/1.67/bcprov-jdk15on-1.67.jar; invalid manifest format
[INFO] /application/target/generated-sources/swagger/src/main/java/uk/org/iay/md/validator/RFC3339DateFormat.java: /application/target/generated-sources/swagger/src/main/java/uk/org/iay/md/validator/RFC3339DateFormat.java uses or overrides a deprecated API.
[INFO] /application/target/generated-sources/swagger/src/main/java/uk/org/iay/md/validator/RFC3339DateFormat.java: Recompile with -Xlint:deprecation for details.
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:testResources (default-testResources) @ md-validator ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 5 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ md-validator ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 1 source file to /application/target/test-classes
[ERROR] error reading /root/.m2/repository/org/bouncycastle/bcprov-jdk15on/1.67/bcprov-jdk15on-1.67.jar; invalid manifest format
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ md-validator ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running uk.org.iay.md.validator.api.ValidatorsApiControllerTest
[ERROR] Tests run: 13, Failures: 9, Errors: 0, Skipped: 4, Time elapsed: 5.795 s <<< FAILURE! - in uk.org.iay.md.validator.api.ValidatorsApiControllerTest
[ERROR] uk.org.iay.md.validator.api.ValidatorsApiControllerTest.springTestContextPrepareTestInstance  Time elapsed: 5.556 s  <<< FAILURE!
java.lang.NoSuchMethodError: org.springframework.test.context.TestContext.computeAttribute(Ljava/lang/String;Ljava/util/function/Function;)Ljava/lang/Object;
[ERROR] uk.org.iay.md.validator.api.ValidatorsApiControllerTest.springTestContextBeforeTestMethod[public void uk.org.iay.md.validator.api.ValidatorsApiControllerTest.testGitHub6() throws java.lang.Exception](0)  Time elapsed: 5.566 s  <<< FAILURE!
java.lang.NoSuchMethodError: org.springframework.test.context.TestContext.computeAttribute(Ljava/lang/String;Ljava/util/function/Function;)Ljava/lang/Object;
[ERROR] uk.org.iay.md.validator.api.ValidatorsApiControllerTest.springTestContextAfterTestMethod[public void uk.org.iay.md.validator.api.ValidatorsApiControllerTest.testGitHub6() throws java.lang.Exception](0)  Time elapsed: 0.01 s  <<< FAILURE!
java.lang.NoSuchMethodError: org.springframework.test.context.TestContext.computeAttribute(Ljava/lang/String;Ljava/util/function/Function;)Ljava/lang/Object;
[ERROR] uk.org.iay.md.validator.api.ValidatorsApiControllerTest.springTestContextBeforeTestMethod[public void uk.org.iay.md.validator.api.ValidatorsApiControllerTest.testSchemaFailure() throws java.lang.Exception](0)  Time elapsed: 0.014 s  <<< FAILURE!
java.lang.NoSuchMethodError: org.springframework.test.context.TestContext.computeAttribute(Ljava/lang/String;Ljava/util/function/Function;)Ljava/lang/Object;
[ERROR] uk.org.iay.md.validator.api.ValidatorsApiControllerTest.springTestContextAfterTestMethod[public void uk.org.iay.md.validator.api.ValidatorsApiControllerTest.testSchemaFailure() throws java.lang.Exception](0)  Time elapsed: 0.002 s  <<< FAILURE!
java.lang.NoSuchMethodError: org.springframework.test.context.TestContext.computeAttribute(Ljava/lang/String;Ljava/util/function/Function;)Ljava/lang/Object;
[ERROR] uk.org.iay.md.validator.api.ValidatorsApiControllerTest.springTestContextBeforeTestMethod[public void uk.org.iay.md.validator.api.ValidatorsApiControllerTest.testTestValidation() throws java.lang.Exception](0)  Time elapsed: 0.005 s  <<< FAILURE!
java.lang.NoSuchMethodError: org.springframework.test.context.TestContext.computeAttribute(Ljava/lang/String;Ljava/util/function/Function;)Ljava/lang/Object;
[ERROR] uk.org.iay.md.validator.api.ValidatorsApiControllerTest.springTestContextAfterTestMethod[public void uk.org.iay.md.validator.api.ValidatorsApiControllerTest.testTestValidation() throws java.lang.Exception](0)  Time elapsed: 0.003 s  <<< FAILURE!
java.lang.NoSuchMethodError: org.springframework.test.context.TestContext.computeAttribute(Ljava/lang/String;Ljava/util/function/Function;)Ljava/lang/Object;
[ERROR] uk.org.iay.md.validator.api.ValidatorsApiControllerTest.springTestContextBeforeTestMethod[public void uk.org.iay.md.validator.api.ValidatorsApiControllerTest.testValidators() throws java.lang.Exception](0)  Time elapsed: 0.006 s  <<< FAILURE!
java.lang.NoSuchMethodError: org.springframework.test.context.TestContext.computeAttribute(Ljava/lang/String;Ljava/util/function/Function;)Ljava/lang/Object;
[ERROR] uk.org.iay.md.validator.api.ValidatorsApiControllerTest.springTestContextAfterTestMethod[public void uk.org.iay.md.validator.api.ValidatorsApiControllerTest.testValidators() throws java.lang.Exception](0)  Time elapsed: 0.007 s  <<< FAILURE!
java.lang.NoSuchMethodError: org.springframework.test.context.TestContext.computeAttribute(Ljava/lang/String;Ljava/util/function/Function;)Ljava/lang/Object;
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ValidatorsApiControllerTest>AbstractTestNGSpringContextTests.springTestContextAfterTestMethod:214 » NoSuchMethod
[ERROR]   ValidatorsApiControllerTest>AbstractTestNGSpringContextTests.springTestContextAfterTestMethod:214 » NoSuchMethod
[ERROR]   ValidatorsApiControllerTest>AbstractTestNGSpringContextTests.springTestContextAfterTestMethod:214 » NoSuchMethod
[ERROR]   ValidatorsApiControllerTest>AbstractTestNGSpringContextTests.springTestContextAfterTestMethod:214 » NoSuchMethod
[ERROR]   ValidatorsApiControllerTest>AbstractTestNGSpringContextTests.springTestContextBeforeTestMethod:161 » NoSuchMethod
[ERROR]   ValidatorsApiControllerTest>AbstractTestNGSpringContextTests.springTestContextBeforeTestMethod:161 » NoSuchMethod
[ERROR]   ValidatorsApiControllerTest>AbstractTestNGSpringContextTests.springTestContextBeforeTestMethod:161 » NoSuchMethod
[ERROR]   ValidatorsApiControllerTest>AbstractTestNGSpringContextTests.springTestContextBeforeTestMethod:161 » NoSuchMethod
[ERROR]   ValidatorsApiControllerTest>AbstractTestNGSpringContextTests.springTestContextPrepareTestInstance:149 » NoSuchMethod
[INFO] 
[ERROR] Tests run: 13, Failures: 9, Errors: 0, Skipped: 4
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 03:48 min
[INFO] Finished at: 2021-02-10T17:28:41Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M5:test (default-test) on project md-validator: There are test failures.
[ERROR] 
[ERROR] Please refer to /application/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
The command '/bin/sh -c mvn --batch-mode     -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn     package' returned a non-zero code: 1
```

**Comment by iay** [comment]
Confirming that I can reproduce this here.

**Comment by iay** [comment]
Spring Boot and Spring Framework have a very close relationship (Spring Boot's test component, in particular, seems to use some non-public APIs in Spring Framework's test component) and they had become different enough to make things break.

I have tried to make the dependencies within my code more explicit to make this less likely in future, but the real solution turned out to be to bump the version of Spring Boot being used.

Fixed by commit 0b2d7b36fa226d4094d87ee6e00540414b250a4f. The image on Docker Hub (`ianayoung/md-validator`) has rebuilt automatically and I have updated my local deployment of `md-validator-ukf` as well.
